addons/sms/models/sms_sms.py

```diff
@@ -83,6 +82,11 @@ def _compute_sms_tracker_id(self):
         for sms in self.filtered(lambda s: s.uuid in tracker_ids_by_sms_uuid):
             sms.sms_tracker_id = tracker_ids_by_sms_uuid[sms.uuid]
 
+    @api.depends('partner_id')
+    def _compute_number(self):
+        for sms in self:
+            sms.number = sms.partner_id.phone
```

Comment:
Usually the correct field will be 'mobile'
Additionally it's probably best to act as if we don't know what the field is, in case it changes

You can use  `_phone_get_number_fields` to get the valid fields to send an sms on a record, and just pick the first valid value if any

Or we could use `_sms_get_recipients_info` which already does that, and a bit more.